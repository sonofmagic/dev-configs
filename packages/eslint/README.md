# @icebreakers/eslint-config

> extend [@antfu/eslint-config](https://www.npmjs.com/package/@antfu/eslint-config) and add `tailwindcss` , `mdx` , and `prettier` format support!

## Usage

```bash
pnpm i -D eslint @icebreakers/eslint-config
```

```ts
// eslint.config.mjs
import { icebreaker } from '@icebreakers/eslint-config'

export default icebreaker()
```

`icebreaker()` returns a `FlatConfigComposer`, so you can opt into extra presets by passing the options documented below and still append workspace-specific rules.

## Available options

```ts
import { icebreaker } from '@icebreakers/eslint-config'

export default icebreaker({
  vue: true,
  typescript: true,
  tailwindcss: {
    tailwindConfig: './tailwind.config.ts',
  },
  mdx: process.env.ENABLE_MDX === 'true',
  a11y: true, // load JSX + Vue accessibility helpers
})
```

- `tailwindcss` – enable Tailwind presets (`true`) or provide entry paths for Tailwind v3/v4.
- `mdx` – enable linting for `.mdx` files with Remark-powered processors.
- `a11y` – adds accessibility rules for whichever of `vue`/`react` you enable via @antfu.
- `vue`, `typescript`, `javascript`, `test` – extend the upstream `@antfu/eslint-config` options; the defaults bundle stricter unused checks and Vue fixes for Ionic/Weapp projects.

## VS Code support

Install [VS Code ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)

Version: ( >=`3.0.10`)
