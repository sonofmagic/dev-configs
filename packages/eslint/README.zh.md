# @icebreakers/eslint-config

## 简介

`@icebreakers/eslint-config` 基于 `@antfu/eslint-config` 的 flat config
预设，额外补充了 Tailwind CSS、MDX、Vue 无障碍以及 Icebreaker
团队常用的 TypeScript 默认规则。它返回一个 `FlatConfigComposer`，
可以按需启用不同预设，并继续追加工作区特定的覆盖项。

## 环境要求

- Node.js 18 或更高版本
- 支持 Flat Config 的 ESLint 9
- 如需启用 Tailwind、MDX、UnoCSS 等，可安装对应的可选依赖：
  `eslint-plugin-tailwindcss` / `eslint-plugin-better-tailwindcss`、
  `eslint-plugin-mdx`、`@unocss/eslint-plugin`

## 安装

```bash
pnpm add -D eslint @icebreakers/eslint-config
```

## 快速上手

在项目根目录创建 `eslint.config.ts`（或 `.mjs`）：

```ts
import { icebreaker } from '@icebreakers/eslint-config'

export default icebreaker()
```

使用包管理器运行 ESLint：

```bash
pnpm eslint \"src/**/*.ts\"
```

如果仍需兼容旧的 `.eslintrc` 流程，可改用 `icebreakerLegacy()`。

## 启用可选预设

所有可选项与 `@antfu/eslint-config` 保持一致，并叠加 Icebreaker 的
调整：

```ts
import { icebreaker } from '@icebreakers/eslint-config'

export default icebreaker({
  vue: true, // 或 { vueVersion: 2 }
  react: true,
  typescript: true,
  test: true,
  tailwindcss: {
    tailwindConfig: './tailwind.config.ts',
  },
  mdx: process.env.LINT_MDX === 'true',
  a11y: true,
  nestjs: true,
  ionic: true,
  weapp: true,
  formatters: true,
})
```

- `vue`：启用 Vue 规则，可根据 Vue 2/3 自动切换，并在 `ionic`、`weapp`
  选项开启时追加对应覆盖。
- `react`：复用上游 React 预设，配合 `a11y` 注入无障碍插件。
- `tailwindcss`：传入 `true` 使用内置 Tailwind flat 配置，或通过对象
  指定 Tailwind v4 的入口文件 / v3 的配置文件路径。
- `mdx`：激活 `eslint-plugin-mdx` 处理 `.mdx` 文件。
- `a11y`：按需引入 JSX 与 Vue 的无障碍规则。
- `typescript`：开启 TypeScript 预设，并在 `nestjs` 为 `true` 时放宽
  NestJS 场景常见限制。
- `formatters`：默认启用格式化辅助规则。
- `test`：放宽 Vitest / Jest 常见规则，例如关闭
  `test/prefer-lowercase-title`。

## 追加自定义配置

`icebreaker()` 返回的 composer 支持继续拼接 Flat Config：

```ts
import { icebreaker } from '@icebreakers/eslint-config'

export default icebreaker(
  { typescript: true },
  {
    files: ['*.vue'],
    rules: {
      'vue/no-undef-components': 'off',
    },
  },
)
```

亦可追加其它第三方或内部预设，统一合并。

## IDE 集成

- VS Code 安装 ESLint 扩展（版本需 ≥ 3.0.10）。
- 老版本 VS Code 需在设置中启用
  `"eslint.experimental.useFlatConfig": true`。
- 在 Git 钩子或 CI 中执行 `pnpm lint -- --fix` 确保格式一致。

## 常见问题

- 如果提示缺少插件，说明当前工作区未安装对应可选依赖，可通过
  `pnpm add -D` 补齐。
- 与旧版 `.eslintrc` 混用时建议先改用 `icebreakerLegacy()`，逐步迁移至
  Flat Config。
- Tailwind 校验依赖 `tailwind.config.*`，Monorepo 或自定义构建路径时请
  确认配置文件位置。
