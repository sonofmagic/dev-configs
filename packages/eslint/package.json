{
  "name": "@icebreakers/eslint-config",
  "type": "module",
  "version": "1.5.8",
  "description": "ESLint preset from Icebreaker's dev-configs",
  "author": "ice breaker <1324318532@qq.com>",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/sonofmagic/dev-configs.git",
    "directory": "packages/eslint"
  },
  "bugs": {
    "url": "https://github.com/sonofmagic/dev-configs/issues"
  },
  "keywords": [
    "dev-configs",
    "eslint-config",
    "icebreaker"
  ],
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.js",
      "require": "./dist/index.cjs"
    },
    "./*": "./*"
  },
  "main": "./dist/index.cjs",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "typesVersions": {
    "*": {
      "*": [
        "./dist/*",
        "./dist/index.d.ts"
      ]
    }
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "dev": "tsup --watch --sourcemap",
    "build": "tsup",
    "test": "vitest run",
    "test:dev": "vitest",
    "release": "pnpm publish",
    "sync": "cnpm sync @icebreakers/eslint-config"
  },
  "dependencies": {
    "@antfu/eslint-config": "6.2.0",
    "@eslint-react/eslint-plugin": "^2.2.4",
    "@next/eslint-plugin-next": "^16.0.1",
    "eslint-plugin-better-tailwindcss": "^3.7.10",
    "eslint-plugin-format": "1.0.2",
    "eslint-plugin-jsx-a11y": "^6.10.2",
    "eslint-plugin-react-hooks": "^7.0.1",
    "eslint-plugin-react-refresh": "^0.4.24",
    "eslint-plugin-tailwindcss": "3.18.2",
    "eslint-plugin-vuejs-accessibility": "^2.4.1"
  },
  "optionalDependencies": {
    "@unocss/eslint-plugin": "66.5.4",
    "eslint-plugin-mdx": "3.6.2"
  },
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  }
}
