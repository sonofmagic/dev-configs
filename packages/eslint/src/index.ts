export * from './factory'
export * from './preset'
export type {
  TailwindcssConfig,
  TailwindcssOption,
  UserConfigItem,
  UserDefinedOptions,
} from './types'
