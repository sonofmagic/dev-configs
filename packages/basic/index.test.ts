import { ESLint } from 'eslint'
import { resolve } from 'path'
describe('basic', () => {
  const eslint = new ESLint()
  it('snapshot', async () => {
    const res = await eslint.calculateConfigForFile(resolve(__dirname, 'index.js'))
    expect(res).toMatchSnapshot()
  })
})
