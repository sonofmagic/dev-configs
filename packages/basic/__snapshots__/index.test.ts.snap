// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
{
  "env": {
    "browser": true,
    "es2021": true,
    "jest/globals": true,
    "node": true,
  },
  "globals": {
    "document": "readonly",
    "navigator": "readonly",
    "window": "readonly",
  },
  "ignorePatterns": [],
  "noInlineConfig": undefined,
  "parser": null,
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true,
    },
    "ecmaVersion": 2022,
    "sourceType": "module",
  },
  "plugins": [
    "promise",
    "n",
    "import",
    "prettier",
    "jest",
  ],
  "reportUnusedDisableDirectives": undefined,
  "rules": {
    "@babel/object-curly-spacing": [
      "off",
    ],
    "@babel/semi": [
      "off",
    ],
    "@typescript-eslint/block-spacing": [
      "off",
    ],
    "@typescript-eslint/brace-style": [
      "off",
    ],
    "@typescript-eslint/comma-dangle": [
      "off",
    ],
    "@typescript-eslint/comma-spacing": [
      "off",
    ],
    "@typescript-eslint/func-call-spacing": [
      "off",
    ],
    "@typescript-eslint/indent": [
      "off",
    ],
    "@typescript-eslint/key-spacing": [
      "off",
    ],
    "@typescript-eslint/keyword-spacing": [
      "off",
    ],
    "@typescript-eslint/lines-around-comment": [
      0,
    ],
    "@typescript-eslint/member-delimiter-style": [
      "off",
    ],
    "@typescript-eslint/no-extra-parens": [
      "off",
    ],
    "@typescript-eslint/no-extra-semi": [
      "off",
    ],
    "@typescript-eslint/object-curly-spacing": [
      "off",
    ],
    "@typescript-eslint/quotes": [
      0,
    ],
    "@typescript-eslint/semi": [
      "off",
    ],
    "@typescript-eslint/space-before-blocks": [
      "off",
    ],
    "@typescript-eslint/space-before-function-paren": [
      "off",
    ],
    "@typescript-eslint/space-infix-ops": [
      "off",
    ],
    "@typescript-eslint/type-annotation-spacing": [
      "off",
    ],
    "accessor-pairs": [
      "error",
      {
        "enforceForClassMembers": true,
        "getWithoutSet": false,
        "setWithoutGet": true,
      },
    ],
    "array-bracket-newline": [
      "off",
    ],
    "array-bracket-spacing": [
      "off",
      "never",
    ],
    "array-callback-return": [
      "error",
      {
        "allowImplicit": false,
        "allowVoid": false,
        "checkForEach": false,
      },
    ],
    "array-element-newline": [
      "off",
    ],
    "arrow-body-style": [
      "off",
    ],
    "arrow-parens": [
      "off",
    ],
    "arrow-spacing": [
      "off",
      {
        "after": true,
        "before": true,
      },
    ],
    "babel/object-curly-spacing": [
      "off",
    ],
    "babel/quotes": [
      0,
    ],
    "babel/semi": [
      "off",
    ],
    "block-spacing": [
      "off",
      "always",
    ],
    "brace-style": [
      "off",
      "1tbs",
      {
        "allowSingleLine": true,
      },
    ],
    "camelcase": [
      "error",
      {
        "allow": [
          "^UNSAFE_",
        ],
        "ignoreDestructuring": false,
        "ignoreGlobals": true,
        "ignoreImports": false,
        "properties": "never",
      },
    ],
    "comma-dangle": [
      "off",
      {
        "arrays": "never",
        "exports": "never",
        "functions": "never",
        "imports": "never",
        "objects": "never",
      },
    ],
    "comma-spacing": [
      "off",
      {
        "after": true,
        "before": false,
      },
    ],
    "comma-style": [
      "off",
      "last",
    ],
    "computed-property-spacing": [
      "off",
      "never",
      {
        "enforceForClassMembers": true,
      },
    ],
    "constructor-super": [
      "error",
    ],
    "curly": [
      0,
      "multi-line",
    ],
    "default-case-last": [
      "error",
    ],
    "dot-location": [
      "off",
      "property",
    ],
    "dot-notation": [
      "error",
      {
        "allowKeywords": true,
        "allowPattern": "",
      },
    ],
    "eol-last": [
      "off",
    ],
    "eqeqeq": [
      "error",
      "always",
      {
        "null": "ignore",
      },
    ],
    "flowtype/boolean-style": [
      "off",
    ],
    "flowtype/delimiter-dangle": [
      "off",
    ],
    "flowtype/generic-spacing": [
      "off",
    ],
    "flowtype/object-type-curly-spacing": [
      "off",
    ],
    "flowtype/object-type-delimiter": [
      "off",
    ],
    "flowtype/quotes": [
      "off",
    ],
    "flowtype/semi": [
      "off",
    ],
    "flowtype/space-after-type-colon": [
      "off",
    ],
    "flowtype/space-before-generic-bracket": [
      "off",
    ],
    "flowtype/space-before-type-colon": [
      "off",
    ],
    "flowtype/union-intersection-spacing": [
      "off",
    ],
    "func-call-spacing": [
      "off",
      "never",
    ],
    "function-call-argument-newline": [
      "off",
    ],
    "function-paren-newline": [
      "off",
    ],
    "generator-star": [
      "off",
    ],
    "generator-star-spacing": [
      "off",
      {
        "after": true,
        "before": true,
      },
    ],
    "implicit-arrow-linebreak": [
      "off",
    ],
    "import/export": [
      "error",
    ],
    "import/first": [
      "error",
    ],
    "import/no-absolute-path": [
      "error",
      {
        "amd": false,
        "commonjs": true,
        "esmodule": true,
      },
    ],
    "import/no-duplicates": [
      "error",
    ],
    "import/no-named-default": [
      "error",
    ],
    "import/no-webpack-loader-syntax": [
      "error",
    ],
    "indent": [
      "off",
      2,
      {
        "ArrayExpression": 1,
        "CallExpression": {
          "arguments": 1,
        },
        "FunctionDeclaration": {
          "body": 1,
          "parameters": 1,
        },
        "FunctionExpression": {
          "body": 1,
          "parameters": 1,
        },
        "ImportDeclaration": 1,
        "MemberExpression": 1,
        "ObjectExpression": 1,
        "SwitchCase": 1,
        "VariableDeclarator": 1,
        "flatTernaryExpressions": false,
        "ignoreComments": false,
        "ignoredNodes": [
          "TemplateLiteral *",
          "JSXElement",
          "JSXElement > *",
          "JSXAttribute",
          "JSXIdentifier",
          "JSXNamespacedName",
          "JSXMemberExpression",
          "JSXSpreadAttribute",
          "JSXExpressionContainer",
          "JSXOpeningElement",
          "JSXClosingElement",
          "JSXFragment",
          "JSXOpeningFragment",
          "JSXClosingFragment",
          "JSXText",
          "JSXEmptyExpression",
          "JSXSpreadChild",
        ],
        "offsetTernaryExpressions": true,
        "outerIIFEBody": 1,
      },
    ],
    "indent-legacy": [
      "off",
    ],
    "jsx-quotes": [
      "off",
    ],
    "key-spacing": [
      "off",
      {
        "afterColon": true,
        "beforeColon": false,
      },
    ],
    "keyword-spacing": [
      "off",
      {
        "after": true,
        "before": true,
      },
    ],
    "linebreak-style": [
      "off",
    ],
    "lines-around-comment": [
      0,
    ],
    "lines-between-class-members": [
      "error",
      "always",
      {
        "exceptAfterSingleLine": true,
      },
    ],
    "max-len": [
      0,
    ],
    "max-statements-per-line": [
      "off",
    ],
    "multiline-ternary": [
      "off",
      "always-multiline",
    ],
    "n/handle-callback-err": [
      "error",
      "^(err|error)$",
    ],
    "n/no-callback-literal": [
      "error",
    ],
    "n/no-deprecated-api": [
      "error",
    ],
    "n/no-exports-assign": [
      "error",
    ],
    "n/no-new-require": [
      "error",
    ],
    "n/no-path-concat": [
      "error",
    ],
    "n/process-exit-as-throw": [
      "error",
    ],
    "new-cap": [
      "error",
      {
        "capIsNew": false,
        "newIsCap": true,
        "properties": true,
      },
    ],
    "new-parens": [
      "off",
    ],
    "newline-per-chained-call": [
      "off",
    ],
    "no-array-constructor": [
      "error",
    ],
    "no-arrow-condition": [
      "off",
    ],
    "no-async-promise-executor": [
      "error",
    ],
    "no-caller": [
      "error",
    ],
    "no-case-declarations": [
      "error",
    ],
    "no-class-assign": [
      "error",
    ],
    "no-comma-dangle": [
      "off",
    ],
    "no-compare-neg-zero": [
      "error",
    ],
    "no-cond-assign": [
      "error",
    ],
    "no-confusing-arrow": [
      0,
    ],
    "no-const-assign": [
      "error",
    ],
    "no-constant-condition": [
      "error",
      {
        "checkLoops": false,
      },
    ],
    "no-control-regex": [
      "error",
    ],
    "no-debugger": [
      "error",
    ],
    "no-delete-var": [
      "error",
    ],
    "no-dupe-args": [
      "error",
    ],
    "no-dupe-class-members": [
      "error",
    ],
    "no-dupe-keys": [
      "error",
    ],
    "no-duplicate-case": [
      "error",
    ],
    "no-empty": [
      "error",
      {
        "allowEmptyCatch": true,
      },
    ],
    "no-empty-character-class": [
      "error",
    ],
    "no-empty-pattern": [
      "error",
    ],
    "no-eval": [
      "error",
    ],
    "no-ex-assign": [
      "error",
    ],
    "no-extend-native": [
      "error",
    ],
    "no-extra-bind": [
      "error",
    ],
    "no-extra-boolean-cast": [
      "error",
    ],
    "no-extra-parens": [
      "off",
      "functions",
    ],
    "no-extra-semi": [
      "off",
    ],
    "no-fallthrough": [
      "error",
    ],
    "no-floating-decimal": [
      "off",
    ],
    "no-func-assign": [
      "error",
    ],
    "no-global-assign": [
      "error",
    ],
    "no-implied-eval": [
      "error",
    ],
    "no-import-assign": [
      "error",
    ],
    "no-invalid-regexp": [
      "error",
    ],
    "no-irregular-whitespace": [
      "error",
    ],
    "no-iterator": [
      "error",
    ],
    "no-labels": [
      "error",
      {
        "allowLoop": false,
        "allowSwitch": false,
      },
    ],
    "no-lone-blocks": [
      "error",
    ],
    "no-loss-of-precision": [
      "error",
    ],
    "no-misleading-character-class": [
      "error",
    ],
    "no-mixed-operators": [
      0,
      {
        "allowSamePrecedence": true,
        "groups": [
          [
            "==",
            "!=",
            "===",
            "!==",
            ">",
            ">=",
            "<",
            "<=",
          ],
          [
            "&&",
            "||",
          ],
          [
            "in",
            "instanceof",
          ],
        ],
      },
    ],
    "no-mixed-spaces-and-tabs": [
      "off",
    ],
    "no-multi-spaces": [
      "off",
    ],
    "no-multi-str": [
      "error",
    ],
    "no-multiple-empty-lines": [
      "off",
      {
        "max": 1,
        "maxBOF": 0,
        "maxEOF": 0,
      },
    ],
    "no-new": [
      "error",
    ],
    "no-new-func": [
      "error",
    ],
    "no-new-object": [
      "error",
    ],
    "no-new-symbol": [
      "error",
    ],
    "no-new-wrappers": [
      "error",
    ],
    "no-obj-calls": [
      "error",
    ],
    "no-octal": [
      "error",
    ],
    "no-octal-escape": [
      "error",
    ],
    "no-proto": [
      "error",
    ],
    "no-prototype-builtins": [
      "error",
    ],
    "no-redeclare": [
      "error",
      {
        "builtinGlobals": false,
      },
    ],
    "no-regex-spaces": [
      "error",
    ],
    "no-reserved-keys": [
      "off",
    ],
    "no-return-assign": [
      "error",
      "except-parens",
    ],
    "no-self-assign": [
      "error",
      {
        "props": true,
      },
    ],
    "no-self-compare": [
      "error",
    ],
    "no-sequences": [
      "error",
    ],
    "no-shadow-restricted-names": [
      "error",
    ],
    "no-space-before-semi": [
      "off",
    ],
    "no-spaced-func": [
      "off",
    ],
    "no-sparse-arrays": [
      "error",
    ],
    "no-tabs": [
      0,
    ],
    "no-template-curly-in-string": [
      "error",
    ],
    "no-this-before-super": [
      "error",
    ],
    "no-throw-literal": [
      "error",
    ],
    "no-trailing-spaces": [
      "off",
    ],
    "no-undef": [
      "error",
    ],
    "no-undef-init": [
      "error",
    ],
    "no-unexpected-multiline": [
      0,
    ],
    "no-unmodified-loop-condition": [
      "error",
    ],
    "no-unneeded-ternary": [
      "error",
      {
        "defaultAssignment": false,
      },
    ],
    "no-unreachable": [
      "error",
    ],
    "no-unreachable-loop": [
      "error",
    ],
    "no-unsafe-finally": [
      "error",
    ],
    "no-unsafe-negation": [
      "error",
    ],
    "no-unused-expressions": [
      "error",
      {
        "allowShortCircuit": true,
        "allowTaggedTemplates": true,
        "allowTernary": true,
        "enforceForJSX": false,
      },
    ],
    "no-unused-vars": [
      "error",
      {
        "args": "none",
        "caughtErrors": "none",
        "ignoreRestSiblings": true,
        "vars": "all",
      },
    ],
    "no-use-before-define": [
      "error",
      {
        "classes": false,
        "functions": false,
        "variables": false,
      },
    ],
    "no-useless-backreference": [
      "error",
    ],
    "no-useless-call": [
      "error",
    ],
    "no-useless-catch": [
      "error",
    ],
    "no-useless-computed-key": [
      "error",
    ],
    "no-useless-constructor": [
      "error",
    ],
    "no-useless-escape": [
      "error",
    ],
    "no-useless-rename": [
      "error",
    ],
    "no-useless-return": [
      "error",
    ],
    "no-var": [
      "warn",
    ],
    "no-void": [
      "error",
    ],
    "no-whitespace-before-property": [
      "off",
    ],
    "no-with": [
      "error",
    ],
    "no-wrap-func": [
      "off",
    ],
    "nonblock-statement-body-position": [
      "off",
    ],
    "object-curly-newline": [
      "off",
      {
        "consistent": true,
        "multiline": true,
      },
    ],
    "object-curly-spacing": [
      "off",
      "always",
    ],
    "object-property-newline": [
      "off",
      {
        "allowAllPropertiesOnSameLine": false,
        "allowMultiplePropertiesPerLine": true,
      },
    ],
    "object-shorthand": [
      "warn",
      "properties",
    ],
    "one-var": [
      "error",
      {
        "initialized": "never",
      },
    ],
    "one-var-declaration-per-line": [
      "off",
    ],
    "operator-linebreak": [
      "off",
      "after",
      {
        "overrides": {
          ":": "before",
          "?": "before",
          "|>": "before",
        },
      },
    ],
    "padded-blocks": [
      "off",
      {
        "blocks": "never",
        "classes": "never",
        "switches": "never",
      },
    ],
    "prefer-arrow-callback": [
      "off",
    ],
    "prefer-const": [
      "error",
      {
        "destructuring": "all",
        "ignoreReadBeforeAssign": false,
      },
    ],
    "prefer-promise-reject-errors": [
      "error",
    ],
    "prefer-regex-literals": [
      "error",
      {
        "disallowRedundantWrapping": true,
      },
    ],
    "prettier/prettier": [
      "error",
    ],
    "promise/param-names": [
      "error",
    ],
    "quote-props": [
      "off",
      "as-needed",
    ],
    "quotes": [
      0,
      "single",
      {
        "allowTemplateLiterals": false,
        "avoidEscape": true,
      },
    ],
    "react/jsx-child-element-spacing": [
      "off",
    ],
    "react/jsx-closing-bracket-location": [
      "off",
    ],
    "react/jsx-closing-tag-location": [
      "off",
    ],
    "react/jsx-curly-newline": [
      "off",
    ],
    "react/jsx-curly-spacing": [
      "off",
    ],
    "react/jsx-equals-spacing": [
      "off",
    ],
    "react/jsx-first-prop-new-line": [
      "off",
    ],
    "react/jsx-indent": [
      "off",
    ],
    "react/jsx-indent-props": [
      "off",
    ],
    "react/jsx-max-props-per-line": [
      "off",
    ],
    "react/jsx-newline": [
      "off",
    ],
    "react/jsx-one-expression-per-line": [
      "off",
    ],
    "react/jsx-props-no-multi-spaces": [
      "off",
    ],
    "react/jsx-space-before-closing": [
      "off",
    ],
    "react/jsx-tag-spacing": [
      "off",
    ],
    "react/jsx-wrap-multilines": [
      "off",
    ],
    "rest-spread-spacing": [
      "off",
      "never",
    ],
    "semi": [
      "off",
      "never",
    ],
    "semi-spacing": [
      "off",
      {
        "after": true,
        "before": false,
      },
    ],
    "semi-style": [
      "off",
    ],
    "space-after-function-name": [
      "off",
    ],
    "space-after-keywords": [
      "off",
    ],
    "space-before-blocks": [
      "off",
      "always",
    ],
    "space-before-function-paren": [
      "off",
      "always",
    ],
    "space-before-function-parentheses": [
      "off",
    ],
    "space-before-keywords": [
      "off",
    ],
    "space-in-brackets": [
      "off",
    ],
    "space-in-parens": [
      "off",
      "never",
    ],
    "space-infix-ops": [
      "off",
    ],
    "space-return-throw-case": [
      "off",
    ],
    "space-unary-ops": [
      "off",
      {
        "nonwords": false,
        "words": true,
      },
    ],
    "space-unary-word-ops": [
      "off",
    ],
    "spaced-comment": [
      "error",
      "always",
      {
        "block": {
          "balanced": true,
          "exceptions": [
            "*",
          ],
          "markers": [
            "*package",
            "!",
            ",",
            ":",
            "::",
            "flow-include",
          ],
        },
        "line": {
          "markers": [
            "*package",
            "!",
            "/",
            ",",
            "=",
          ],
        },
      },
    ],
    "standard/array-bracket-even-spacing": [
      "off",
    ],
    "standard/computed-property-even-spacing": [
      "off",
    ],
    "standard/object-curly-even-spacing": [
      "off",
    ],
    "switch-colon-spacing": [
      "off",
    ],
    "symbol-description": [
      "error",
    ],
    "template-curly-spacing": [
      "off",
      "never",
    ],
    "template-tag-spacing": [
      "off",
      "never",
    ],
    "unicode-bom": [
      "error",
      "never",
    ],
    "unicorn/empty-brace-spaces": [
      "off",
    ],
    "unicorn/no-nested-ternary": [
      "off",
    ],
    "unicorn/number-literal-case": [
      "off",
    ],
    "unicorn/template-indent": [
      0,
    ],
    "use-isnan": [
      "error",
      {
        "enforceForIndexOf": true,
        "enforceForSwitchCase": true,
      },
    ],
    "valid-typeof": [
      "error",
      {
        "requireStringLiterals": true,
      },
    ],
    "vue/array-bracket-newline": [
      "off",
    ],
    "vue/array-bracket-spacing": [
      "off",
    ],
    "vue/array-element-newline": [
      "off",
    ],
    "vue/arrow-spacing": [
      "off",
    ],
    "vue/block-spacing": [
      "off",
    ],
    "vue/block-tag-newline": [
      "off",
    ],
    "vue/brace-style": [
      "off",
    ],
    "vue/comma-dangle": [
      "off",
    ],
    "vue/comma-spacing": [
      "off",
    ],
    "vue/comma-style": [
      "off",
    ],
    "vue/dot-location": [
      "off",
    ],
    "vue/func-call-spacing": [
      "off",
    ],
    "vue/html-closing-bracket-newline": [
      "off",
    ],
    "vue/html-closing-bracket-spacing": [
      "off",
    ],
    "vue/html-end-tags": [
      "off",
    ],
    "vue/html-indent": [
      "off",
    ],
    "vue/html-quotes": [
      "off",
    ],
    "vue/html-self-closing": [
      0,
    ],
    "vue/key-spacing": [
      "off",
    ],
    "vue/keyword-spacing": [
      "off",
    ],
    "vue/max-attributes-per-line": [
      "off",
    ],
    "vue/max-len": [
      0,
    ],
    "vue/multiline-html-element-content-newline": [
      "off",
    ],
    "vue/multiline-ternary": [
      "off",
    ],
    "vue/mustache-interpolation-spacing": [
      "off",
    ],
    "vue/no-extra-parens": [
      "off",
    ],
    "vue/no-multi-spaces": [
      "off",
    ],
    "vue/no-spaces-around-equal-signs-in-attribute": [
      "off",
    ],
    "vue/object-curly-newline": [
      "off",
    ],
    "vue/object-curly-spacing": [
      "off",
    ],
    "vue/object-property-newline": [
      "off",
    ],
    "vue/operator-linebreak": [
      "off",
    ],
    "vue/quote-props": [
      "off",
    ],
    "vue/script-indent": [
      "off",
    ],
    "vue/singleline-html-element-content-newline": [
      "off",
    ],
    "vue/space-in-parens": [
      "off",
    ],
    "vue/space-infix-ops": [
      "off",
    ],
    "vue/space-unary-ops": [
      "off",
    ],
    "vue/template-curly-spacing": [
      "off",
    ],
    "wrap-iife": [
      "off",
      "any",
      {
        "functionPrototypeMethods": true,
      },
    ],
    "wrap-regex": [
      "off",
    ],
    "yield-star-spacing": [
      "off",
      "both",
    ],
    "yoda": [
      "error",
      "never",
    ],
  },
  "settings": {},
}
`;
